test/test-framework-spec.md

```diff
@@ -343,25 +404,42 @@ Using the available features, the user can stub with preferred behaviors for fun
     test:prepare(mockClient).when("functionName").doNothing();
     ```
 
-3. Provide a return value.
+3. Provide a return value. The function call will return the specified value, regardless of the arguments' values.
 
     ```ballerina
     test:prepare(mockClient).when("functionName").thenReturn(5);
     ```
 
-4. Provide a return value based on the input.
+4. Provide a return value based on the arguments.  
 
     ```ballerina
     test:prepare(mockClient).when("functionName").withArguments(anydata...).thenReturn(5);
     ```
-   
+
 5. Mock the member variables of an object.
 
     ```ballerina
     test:prepare(mockClient).getMember("member").thenReturn(5);
     ```
 
-#### Case B : Features available in function mocking 
+6. Provide generalized inputs to accept any value for certain arguments.
+
+   This feature can be used when there is a need to selectively differentiate the mock behavior for a set of arguments.
```

Comment:
Do we need this explanation? The spec doesn't need to explain the use case isn't it? It can be added to the learn docs if it is missing.